### DISCUSSION TITLE
Slotted elements

### DISCUSSION THREAD

**Discussion by esaramago** [withastro/roadmap#821] [Proposal]
### Body

# Summary

Implement ::slotted(*), like in WebComponents

# Background & Motivation

For now, I think the only solution to style elements inside <slot> is using `<style is:inline>` to make the style global:
https://stackoverflow.com/questions/75090577/how-to-style-slot-elements-in-astro-with-tailwind

There should be a way to style slotted elements without making the styles global.
In WebComponents we can use ::slotted(*) to style slotted elements. Maybe there should be a similar solution in Astro.

# Goals

- With this solution, we could style elements inside <slot> without using global styles inside components.

# Example

```
.component ::slotted(.child-element) {
   /* styles */
}
```

**Comment by delucis** [comment]
This is possible with the [`:global()` selector](https://docs.astro.build/en/guides/styling/#global-styles):

```css
.component :global(.child-element) {
   /* styles */
}
```
